### PR TITLE
Fix trigger shown and hidden events for dropdown

### PR DESCRIPTION
**Pull request by Johann-S**
Hi,

it's a fix for https://github.com/twbs/bootstrap/issues/16828

Thanks to @DaGLiMiOuX

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 98ccf83677b997b86c7ca141802e90a6e023ed94
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/72183457

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: js/dropdown.js [hunk 1]
```diff
@@ -51,7 +51,7 @@
       if (e.isDefaultPrevented()) return
 
       $this.attr('aria-expanded', 'false')
-      $parent.removeClass('open').trigger('hidden.bs.dropdown', relatedTarget)
+      $parent.removeClass('open').trigger(e = $.Event('hidden.bs.dropdown', relatedTarget))
```
Comment (cvrebert) [inline js/dropdown.js]: The `e = ` is unnecessary.

File: js/dropdown.js [hunk 2]
```diff
@@ -85,7 +85,7 @@
 
       $parent
         .toggleClass('open')
-        .trigger('shown.bs.dropdown', relatedTarget)
+        .trigger(e = $.Event('shown.bs.dropdown', relatedTarget))
```
Comment (cvrebert) [inline js/dropdown.js]: The `e = ` is unnecessary.

File: js/tests/unit/dropdown.js [hunk 3]
```diff
@@ -286,6 +286,38 @@ $(function () {
...
+        + '</ul>'
+    var $dropdown = $(dropdownHTML)
+      .appendTo('#qunit-fixture')
+      .find('[data-toggle="dropdown"]')
+      .bootstrapDropdown()
+    var done = assert.async()
+
+    $dropdown.parent('.dropdown')
+      .on('shown.bs.dropdown', function (e) {
+        assert.notStrictEqual(typeof e, 'undefined');
```
Comment (cvrebert) [inline js/tests/unit/dropdown.js]: Needs to assert that `e.relatedTarget` equals the correct element

File: js/tests/unit/dropdown.js [hunk 4]
```diff
@@ -286,6 +286,38 @@ $(function () {
...
+      })
+      .on('hidden.bs.dropdown', function (e) {
+        assert.notStrictEqual(typeof e, 'undefined')
```
Comment (cvrebert) [inline js/tests/unit/dropdown.js]: Needs to assert that `e.relatedTarget` equals the correct element

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 6aafcfb68441baffd60bfecb59524f26305d865c
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/72423124

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by Johann-S** [comment]
Thanks @cvrebert for your feedbacks :shipit:

**Comment by cvrebert** [comment]
Thanks!

**Pull request closed without merging**